cumulusci/core/config/OrgConfig.py

```diff
@@ -23,15 +24,18 @@ class OrgConfig(BaseConfig):
     # make sure it can be mocked for tests
     SalesforceOAuth2 = SalesforceOAuth2
 
-    def __init__(self, config, name):
+    def __init__(self, config, name, keychain=None, global_org=False):
+        self.keychain = keychain
+        self.global_org = global_org
+
         self.name = name
         self._community_info_cache = {}
         self._client = None
         self._latest_api_version = None
         self._installed_packages = None
         super(OrgConfig, self).__init__(config)
 
-    def refresh_oauth_token(self, keychain, connected_app=None):
+    def refresh_oauth_token(self, keychain, connected_app=None, save=False):
```

Comment:
It would be a cleaner separation of concerns to keep the save logic out of `refresh_oauth_token`. `_save_changes_if` could be renamed to `save_if_changed` and used like this in cases where saving is desired:

    with org_config.save_if_changed():
        org_config.refresh_oauth_token(...)